File: rusty_monopoly/src/dice.rs
use rand::{self, Rng};

pub struct RollResult {
    total: u8,
    is_double: bool,
    is_snake_eyes: bool,
}

pub fn roll_dice() -> RollResult {
    let mut generator = rand::rng();
    let die1 = generator.random_range(1..=6);
    let die2 = generator.random_range(1..=6);

    let total = die1 + die2;
    let is_double = die1 == die2;
    let is_snake_eyes = die1 == 1 && die2 == 1;

    RollResult {
        total,
        is_double,
        is_snake_eyes,
    }
}

File: new_monopoly/src/main/java/com/example/Board/Dice.java
package com.example.Board;

import java.util.Random;

public class Dice {
    private final Random random;

    public Dice() {
        this.random = new Random();
    }

    public RollResults roll() {
        int die1 = random.nextInt(6) + 1;
        int die2 = random.nextInt(6) + 1;
        return new RollResults(die1, die2);
    }

    public class RollResults {
        final int total;
        final boolean isDouble;
        final boolean isSnakeEyes;

        public RollResults(int die1, int die2) {
            this.total = die1 + die2;
            this.isDouble = die1 == die2;
            this.isSnakeEyes = die1 == 1 && die2 == 1;
        }

        public int getTotal() {
            return total;
        }

        public boolean isDouble() {
            return isDouble;
        }

        public boolean isSnakeEyes() {
            return isSnakeEyes;
        }
    }
}